File: 0088-merge-sorted-array/0088-merge-sorted-array.c
void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n) {
    int i = 0;
    int j = 0;
    int arr3[nums1Size + nums2Size];
    int index = 0;

    while (i < m && j < n) {
        if (nums1[i] <= nums2[j]) {
            arr3[index] = nums1[i];
            i++;
        } else {
            arr3[index] = nums2[j];
            j++;
        }
        index++;
    }

    while (i < m) {
        arr3[index] = nums1[i];
        index++;
        i++;
    }

    while (j < n) {
        arr3[index] = nums2[j];
        index++;
        j++;
    }

    for (i = 0; i < m + n; i++) {
        nums1[i] = arr3[i];
    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] ans = new int[m + n];

        int first = 0;
        int second = 0;
        int k = 0;

        while (first < m && second < n) {
            if (nums1[first] <= nums2[second]) {
                ans[k++] = nums1[first++];
            } else {
                ans[k++] = nums2[second++];
            }
        }

        while (first < m) {
            ans[k++] = nums1[first++];
        }

        while (second < n) {
            ans[k++] = nums2[second++];
        }

        for (int i = 0; i < m + n; i++) {
            nums1[i] = ans[i];
        }
    }
}